packages/pigeon/lib/cpp_generator.dart

```diff
@@ -171,6 +171,220 @@ template<class T> class ErrorOr {
 ''');
 }
 
+/// Writes the declaration for the custom class [klass].
+///
+/// See [_writeDataClassImplementation] for the corresponding declaration.
+/// This is intended to be added to the header.
+void _writeDataClassDeclaration(Indent indent, Class klass, Root root,
```

Comment:
I moved these without changes as the very first commit, so excluding the first commit from the review will make it easier to see the substantive changes.